application/fin/fin.go

```diff
@@ -47,16 +47,22 @@ type Report interface {
 	getReference(Transaction) string
 }
 
-func NewBatch(reportFormat string, date time.Time) Report {
+func NewBatch(reportFormat, reportType string, date time.Time) Report {
 	batchDesc := fmt.Sprintf("%s %s JE", date.Format("January 2006"), domain.Env.AppName)
 
 	switch reportFormat {
 	case ReportFormatNetSuite:
+		annual := 0
+		if reportType == "Annual" {
+			annual = 1
+		}
```

Comment:
I think a boolean is better. Also, "Annual" should be a const since it is a critical value in a comparison. You could even change `reportType string` to `isAnnual bool`. If you think you need more flexibility, make a `map` that converts `reportType` to a single-character type code that gets inserted into the transaction ID (row ID?). As it is, a future change that assigns a value > 9 would cause the transaction ID to increase in length. That seems to be a bad thing since it appears to be a conglomeration of a bunch of information packed into a short string with no delimiters.
```suggestion
		annual := false
		if reportType == "Annual" {
			annual = true
		}
```